src/synapse/authProvider.js

```diff
@@ -3,6 +3,11 @@ import { fetchUtils } from "react-admin";
 const authProvider = {
   // called when the user attempts to log in
   login: ({ base_url, username, password }) => {
+    // force homeserver for protection in case the form is manipulated
+    base_url = process.env.REACT_APP_SERVER
+      ? process.env.REACT_APP_SERVER
+      : base_url;
```

Comment:
```suggestion
    base_url = process.env.REACT_APP_SERVER || base_url;
```
See other suggestions.